### PR TITLE
keep going with other test suite runs when failure with one

### PR DESCRIPTION
**Pull request by Rajalakshmi-Girish**
The current logic in [test.sh](https://github.com/etcd-io/etcd/blob/main/scripts/test.sh#L644:#L659) exits the run when there is first failure in the suite names passed to PASSES variable.

This is leading to incomplete results when the user intends to run multiple test suites.
This change continues with the for loop when there is failure in any suite and exits after the completion of all test suite runs.

### CODE REVIEW

**Comment by Rajalakshmi-Girish** [comment]
@serathius @jmhbnz Can you please take a look at this PR?

**Review by jmhbnz — COMMENTED** [review]
Is continuing with other test suites actually what we want though?  

I think it might depend on the situation as when I am developing locally and I run `make test` which runs a bunch of suites if there is a failure with unit tests my preference is to have it stop immediately rather than continue on trying to run other tests which will waste time.

If I'm misunderstanding something about this change please let me know, also curious what other folks think here.

File: scripts/test.sh [hunk 1]
```diff
@@ -646,16 +646,26 @@ function run_pass {
...
   log_callout -e "\\n'${pass}' started at $(date)"
   if "${pass}_pass" "$@" ; then
-    log_success "'${pass}' completed at $(date)"
+    log_success "'${pass}' PASSED and completed at $(date)"
+    return 0
   else
-    log_error "FAIL: '${pass}' failed at $(date)"
-    exit 255
+    log_error "FAIL: '${pass}' FAILED and completed at $(date)"
+    return 2
```
Comment (chaochn47) [inline scripts/test.sh:653]: Is it possible to make this gated by environment variables so you can turn it on in your environment?

Comment (jmhbnz) [reply scripts/test.sh:653]: Great suggestion, this approach sounds good to me.

Comment (Rajalakshmi-Girish) [reply scripts/test.sh]: I have added the gate by the variable `PASSES_CONTINUE`.
Can you please check now?

**Review by jmhbnz — APPROVED** [review]
LGTM

**Review by chaochn47 — APPROVED** [review]

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @Rajalakshmi-Girish

**Pull request merged**